lib/spack/spack/test/packaging.py

```diff
@@ -0,0 +1,252 @@
+
+
+def fake_fetchify(url, pkg):
+    """Fake the URL for a package so it downloads from a file."""
+    fetcher = FetchStrategyComposite()
+    fetcher.append(URLFetchStrategy(url))
+    pkg.fetcher = fetcher
+
+
+@pytest.mark.usefixtures('install_mockery', 'testing_gpg_directory')
+def test_packaging(mock_archive, tmpdir):
+    # tweak patchelf to only do a download
+    spec = Spec("patchelf")
+    spec.concretize()
+    pkg = spack.repo.get(spec)
+    fake_fetchify(pkg.fetcher, pkg)
+    mkdirp(join_path(pkg.prefix, "bin"))
+    patchelfscr = join_path(pkg.prefix, "bin", "patchelf")
+    f = open(patchelfscr, 'w')
+    body = """#!/bin/bash
+echo $PATH"""
+    f.write(body)
+    f.close()
+    st = os.stat(patchelfscr)
+    os.chmod(patchelfscr, st.st_mode | stat.S_IEXEC)
+
+    # Install the test package
+    spec = Spec('trivial-install-test-package')
+    spec.concretize()
+    assert spec.concrete
+    pkg = spack.repo.get(spec)
+    fake_fetchify(mock_archive.url, pkg)
+    pkg.do_install()
+
+    # Put some non-relocatable file in there
+    filename = join_path(spec.prefix, "dummy.txt")
+    with open(filename, "w") as script:
+        script.write(spec.prefix)
+
+    # Create the build cache  and
+    # put it directly into the mirror
+
+    mirror_path = join_path(tmpdir, 'test-mirror')
+    specs = [spec]
+    spack.mirror.create(
+        mirror_path, specs, no_checksum=True
+    )
+
+    parser = argparse.ArgumentParser()
+    buildcache.setup_parser(parser)
+    shutil.copyfile(spack.mock_gpg_keys_path + '/external.key',
+                    'external.key')
+    # Create a private key to sign package with if gpg2 available
+    if has_gnupg2():
+        spack.util.gpg.Gpg.create(name='test key 1', expires='0',
+                                  email='spack@googlegroups.com',
+                                  comment='Spack test key')
+    # Create a build cache with signing
+    args = parser.parse_args(['create', '-d', mirror_path, '-y', str(spec)])
+    buildcache.buildcache(parser, args)
+
+    # Create a build cache without signing, making rpaths relative first
+    # overwriting previous build cache
+    args = parser.parse_args(
+        ['create', '-d', mirror_path, '-f', '-r', '-y', str(spec)])
+    buildcache.buildcache(parser, args)
+
+    # register mirror with spack config
+    mirrors = {'spack-mirror-test': 'file://' + mirror_path}
+    spack.config.update_config('mirrors', mirrors)
+
+    # Uninstall the package
+    pkg.do_uninstall(force=True)
+
+    # download and install tarball
+    file = bindist.download_tarball(spec)
+    bindist.extract_tarball(spec, file, True, True)
+    spack.store.db.reindex(spack.store.layout)
+
+    # Validate the relocation information
+    buildinfo = bindist.read_buildinfo_file(spec.prefix)
+    assert(buildinfo['relocate_textfiles'] == ['dummy.txt'])
+
+    args = parser.parse_args(['install', '-f', '-y', str(spec)])
+    buildcache.install_tarball(spec, args)
```

Comment:
An example of verification in this case is to ensure that Spack thinks the package is installed.